sqlparser/lib/src/engine/module/rtree.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:sqlparser/sqlparser.dart';
+
+class RTreeExtension implements Extension {
+  const RTreeExtension();
+
+  @override
+  void register(SqlEngine engine) {
+    engine.registerModule(_RTreeModule());
+  }
+}
+
+class _RTreeModule extends Module {
+  _RTreeModule() : super('rtree');
+
+  @override
+  Table parseTable(CreateVirtualTableStatement stmt) {
+    final columnNames = stmt.argumentContent;
+
+    if (columnNames.length < 3 || columnNames.length > 11) {
+      throw ArgumentError(
+          'An rtree virtual table is supposed to have between 3 and 11 columns');
+    }
+
+    if (columnNames.length.isEven) {
+      throw ArgumentError(
+          'The rtree has not been initialized with a proper dimension. '
+          'Required is an index, followed by a even number of min/max pairs');
+    }
```

Comment:
Let's also add a testcase for these two errors (they will be rethrown by the `SchemaFromCreateTable` reader), so just calling that and expecting the argument error is enough.

Reply:
done